asg/src/expression/binary.rs

```diff
@@ -176,10 +176,7 @@ impl FromAst<leo_ast::BinaryExpression> for BinaryExpression {
             BinaryOperationClass::Numeric => match left_type {
                 Some(Type::Integer(_)) => (),
                 Some(Type::Group) | Some(Type::Field)
-                    if value.op == BinaryOperation::Add || value.op == BinaryOperation::Sub =>
-                {
```

Comment:
this conflicts with `cargo fmt` and infinite loops. rebase on latest asg-asg -- I ignored this clippy lint here
.